scripts/validate-cloudtrail-logs.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+##? Usage: validate-cloudtrail-logs.sh [options] ACCOUNT START_FROM [UP_TO]
+##?
+##? Options:
+##?   --config=FILE  Path to file containing details of trails (default: $DM_CREDENTIALS_REPO/jenkins-vars/cloudtrail.yaml)
+##?   -v, --verbose
+##?   -n, --dry-run
+##?   -h, --help
+
+set -eo pipefail
+
+help=$(grep "^##?" "$0" | cut -c 5-)
+eval "$(docopts -h "$help" : "$@")"
+
+VERBOSE="${verbose:-$VERBOSE}"
+$VERBOSE && set -x
+
+[ -n "$config" ] && CONFIG_FILE="$config" || CONFIG_FILE="$DM_CREDENTIALS_REPO/jenkins-vars/cloudtrail.yaml"
+[ -f "$CONFIG_FILE" ] || { echo "error: config file '$CONFIG_FILE' does not exist"; exit 64; }
+
+[ -n "$UP_TO" ] || UP_TO="now"
+START_TIME="$(date --date="$START_FROM" --utc --iso-8601=minutes)" || exit 64
+END_TIME="$(date --date="$UP_TO" --utc --iso-8601=minutes)" || exit 64
+
+export ACCOUNT
+account="$("$DM_CREDENTIALS_REPO"/sops-wrapper -d --extract '["cloudtrail_validate_logs_roles"]' "$CONFIG_FILE" | yq 'map(select(.account == env.ACCOUNT))[0]')"
```

Comment:
So if we added these perms to the jenkins profile instance and the infrastructure/ user tolese we could forego all this sops decryption

Reply:
As well as the profile details [the file we're decrypting](https://github.com/alphagov/digitalmarketplace-credentials/pull/221/files#diff-1f8bec20b12553ad99d4b3ed1da377ce) contains the trail arn, which the aws cli needs. I guess we could forgo that file if we rely on the name scheme being consistent.

That being said, I don't know how it would be possible to have the permissions for validating logs from another account in the Jenkins instance profile.